python/sglang/srt/mem_cache/sparsity/algorithms/deepseek_nsa.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import nvtx
+import torch
+
+from sglang.srt.mem_cache.sparsity.algorithms.base_algorithm import (
+    BaseSparseAlgorithm,
+    SparseMode,
+)
+
+if TYPE_CHECKING:
+    from sglang.srt.layers.attention.nsa.nsa_indexer import Indexer
+    from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+logger = logging.getLogger(__name__)
+
+
+class DeepSeekNSAAlgorithm(BaseSparseAlgorithm):
+    """Sparse attention algorithm for DeepSeek NSA."""
+
+    def __init__(self, config, device: torch.device, **kwargs):
+        super().__init__(config, device, **kwargs)
+        self.index_topk = getattr(config, "index_topk", 2048)
+        logger.info(f"DeepSeekNSAAlgorithm initialized: index_topk={self.index_topk}")
+
+    def get_sparse_mode(self) -> SparseMode:
+        return SparseMode.ORIGINAL_WISE
```

Comment:
let's use a clearer name for it, if it's block wise then it's block wise, or specifically dpsk-block-wise